### ISSUE TITLE
Multibuild doesn't work if `vavite` is included within another plugin/nested array

### ISSUE THREAD

**Issue by wrapperup** [cyco130/vavite#94]
I was trying to include vavite as part of a plugin. However, multibuild doesn't seem to hook the `buildStepStart` and `buildStepEnd` in nested plugins, causing vavite to fail building when using `buildSteps`

To reproduce, simply nest vavite in another `PluginOption` array, and include build steps:

```js
defineConfig({
  buildSteps: [
    ...
  ],
  plugins: [
    [
      vavite({
        ...
      })
    ],
  ],
});
```

The plugin loads, but the multibuild hooks are never called, which leads to this error:
```
Error: vavite: You have multiple build steps defined in your Vite config, please use the 'vavite' command instead of 'vite build' to build.
```

**Comment by cyco130** [comment]
The multibuild feature only works through the `vavite` CLI program currently, not with `vite build`. Are you sure you're using it? Alternatively, you can build your own CLI wrapper (e.g. like Rakkas does).

If you are already using it, I need a reproduction because I know it works for Rakkas and Hattip, so it must be something else.

**Comment by wrapperup** [comment]
> The multibuild feature only works through the `vavite` CLI program currently, not with `vite build`. Are you sure you're using it? Alternatively, you can build your own CLI wrapper (e.g. like Rakkas does).
> 
> If you are already using it, I need a reproduction because I know it works for Rakkas and Hattip, so it must be something else.

Yeah I'm using the vavite CLI to run this.

I made a super minimal repro https://github.com/wrapperup/vavite-multibuild-issue

**Comment by cyco130** [comment]
Thanks for the repro, I found the issue: The types are wrong (vavite's fault).

vavite itself is already an array of plugins so the return type should be `Plugin[]` instead of `PluginOption`. `PluginOption` should only be used on the receiving side (e.g. Vite itself).

I will release a fix in the first opportunity but you can use `...vavite()` (instead of `vavite()`) and ignore the type error in the meantime.

**Comment by cyco130** [comment]
Fixed in #96 and published as 4.1.1.
